snapcraft/parts/lifecycle.py

```diff
@@ -186,6 +186,20 @@ def _run_command(  # noqa PLR0913 # pylint: disable=too-many-branches, too-many-
         lifecycle.clean(part_names=part_names)
         return
 
+    if any(
+        "enable-patchelf" in _part.get("build-attributes", [])
+        for _part in getattr(project, "parts", {}).values()
+    ) and any(
+        _part.get("override-prime", None)
+        for _part in getattr(project, "parts", {}).values()
+    ):
+        emit.progress(
+            "Warning: 'enable-patchelf' feature is unsafe if 'override-prime' "
```

Comment:
Just so I am clear, is the following statement correct?

In a project with multiple parts:
- `patchelf` will not run on files primed by parts with `override-prime` defined
- `patchelf` will run on files primed by parts without `override-prime` defined

I would slightly reword this message - 'unsafe' makes me think that the feature is expected to work but experimental. Assuming the statement above is true, perhaps something like:
```
Warning: 'enable-patchelf' feature will not apply to files primed by parts that use 
the 'override-prime' keyword. It's not possible to track file changes in the prime directory.
```

Reply:
> Just so I am clear, is the following statement correct?
> 
> In a project with multiple parts:
> 
> * `patchelf` will not run on files primed by parts with `override-prime` defined
> * `patchelf` will run on files primed by parts without `override-prime` defined

Yes, the statement is correct for now, bug we may change it to something else very soon. So I don't know if I should say it will not work or it may broke under certain circumstances.

Reply:
I wouldn't change the wording based on how soon the fix is coming.  I think it needs to be clearer that snapcraft is not 'unsafe' but 'will not work as expected'

Reply:
OK, updated